core/src/main/java/tc/oc/pgm/stats/StatsMatchModule.java

```diff
@@ -0,0 +1,492 @@
+
+    return task;
+  }
+
+  @EventHandler
+  public void onMatchEnd(MatchFinishEvent event) {
+
+    if (allPlayerStats.isEmpty()) return;
+
+    Map<UUID, Integer> allKills = new HashMap<>();
+    Map<UUID, Integer> allKillstreaks = new HashMap<>();
+    Map<UUID, Integer> allDeaths = new HashMap<>();
+    Map<UUID, Integer> allBowshots = new HashMap<>();
+    Map<UUID, Double> allDamage = new HashMap<>();
+
+    for (Map.Entry<UUID, PlayerStats> mapEntry : allPlayerStats.entrySet()) {
+      UUID playerUUID = mapEntry.getKey();
+      PlayerStats playerStats = mapEntry.getValue();
+
+      getPlayerStat(playerUUID);
+
+      allKills.put(playerUUID, playerStats.kills);
+      allKillstreaks.put(playerUUID, playerStats.killstreakMax);
+      allDeaths.put(playerUUID, playerStats.deaths);
+      allBowshots.put(playerUUID, playerStats.longestBowKill);
+      allDamage.put(playerUUID, playerStats.damageDone);
+    }
+
+    Component killMessage = getMessage("match.stats.kills", sortStats(allKills), TextColor.GREEN);
+    Component killstreakMessage =
+        getMessage("match.stats.killstreak", sortStats(allKillstreaks), TextColor.GREEN);
+    Component deathMessage = getMessage("match.stats.deaths", sortStats(allDeaths), TextColor.RED);
+    Map.Entry<UUID, Integer> bestBowshot = sortStats(allBowshots);
+    if (bestBowshot.getValue() == 1)
+      bestBowshot.setValue(2); // Avoids translating "1 block" vs "n blocks"
+    Component bowshotMessage = getMessage("match.stats.bowshot", bestBowshot, TextColor.YELLOW);
+    Component damageMessage =
+        getMessage("match.stats.damage", sortStatsDouble(allDamage), TextColor.GREEN);
+
+    match
+        .getExecutor(MatchScope.LOADED)
+        .schedule(
+            () -> {
+              for (MatchPlayer viewer : match.getPlayers()) {
+                if (viewer.getSettings().getValue(SettingKey.STATS) == SettingValue.STATS_OFF)
+                  continue;
+
+                viewer.sendMessage(
+                    TextFormatter.horizontalLineHeading(
+                        viewer.getBukkit(),
+                        TranslatableComponent.of("match.stats.title", TextColor.YELLOW),
+                        TextColor.WHITE));
+                viewer.sendMessage(killMessage);
+                viewer.sendMessage(killstreakMessage);
+                viewer.sendMessage(deathMessage);
+                if (bestBowshot.getValue() != 0) viewer.sendMessage(bowshotMessage);
+                viewer.sendMessage(damageMessage);
+                if (verboseStats) viewer.getInventory().setItem(7, createToolItem(viewer));
+              }
+            },
+            5 + 1, // NOTE: This is 1 second after the votebook appears
+            TimeUnit.SECONDS);
+
+    final Collection<Competitor> competitors = match.getCompetitors();
+    // TODO: Support all Parties, Tributes(FFA)
+    if (verboseStats && competitors.stream().noneMatch(c -> c instanceof Tribute)) {
```

Comment:
In reality, you just support teams not competitors, do a check on team match module existing or competitors being Team, instead of checking for no tributes